index.js

```diff
@@ -9,20 +9,60 @@ let validator = require('validator');
 const TAB = '    ';
 const helpRegex = /^\[(string|number|boolean|json)\].*$/i;
 
-function recurse(obj, counter, path) {
+function appendExtraData(tmpl, extra) {
+	var returnObject = {};
+	if (!extra) extra = {};
+	for (let key in tmpl) {
+		if (tmpl.hasOwnProperty(key)) {
+			if (_.isObject(tmpl[key])) {
+				returnObject[key] = appendExtraData(tmpl[key], extra[key]);
+			} else {
+				returnObject[key] = tmpl[key];
+			}
+		}
+	}
+	for (let key in extra) {
+		if (extra.hasOwnProperty(key)) {
+			if (_.isObject(extra[key])) {
+				if (!tmpl[key]) tmpl[key] = {};
+				if (!tmpl[key].toString().match(/^\[json\].*/))
+					returnObject[key] = appendExtraData(tmpl[key], extra[key]);
+			} else {
+				if (!tmpl.hasOwnProperty(key)) returnObject[key] = extra[key];
+			}
+		}
+	}
+	return returnObject;
+}
+
+function setDefaultValues(parsedTmpl, values) {
+	for (var line,i=0; line=parsedTmpl.lines[i]; i++) {
+		var obj = values;
+		for (var k,j=0; k=line.path[j]; j++) obj = obj[k] || '';
+		line.value = obj.toString();
+	}
+}
+
+function interpretTmpl(obj, counter, path) {
 	counter = counter || 1;
 	path = path || [];
 
+	let value = '';
+
 	let returnObject = {
 		lines: [],
 		counter: counter
 	};
+
 	if(!_.isObject(obj)) {
 		let match = null;
 		let type = 'string';
 		
-		if(!_.isString(obj)) {
-			obj = obj.toString();
+		if(!_.isString(obj) || !obj.match(helpRegex)) {
```

Comment:
It seems like this work is more suited to be done in the appendExtraData function. This way our template doesn't have a mix of instructions strings (eg. '[string] Defaults to "The environment."') and value strings (eg. 'The environment.')